### PR TITLE
support multiple non-consecutive intervals in outer query of nested group-by

### PR DESCRIPTION
**Pull request by himanshug**
currently it fails with, "Null storage adapter found. Probably trying to issue a query against a segment being memory unmapped."

### CODE REVIEW

File: processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java [hunk 1]
```diff
@@ -173,14 +175,26 @@ public GroupByQueryQueryToolChest(
...
-      IncrementalIndex index = makeIncrementalIndex(innerQuery, subqueryResult);
+      final IncrementalIndex index = makeIncrementalIndex(innerQuery, subqueryResult);
 
       return new ResourceClosingSequence<>(
           outerQuery.applyLimit(
-              engine.process(
-                  outerQuery,
-                  new IncrementalIndexStorageAdapter(
-                      index
+              Sequences.concat(
```
Comment (xvrl) [inline processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java]: a comment would be nice here to explain why this is done.

Comment (himanshug) [inline processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java]: comment added.

**Comment by nishantmonu51** [comment]
:+1:

**Comment by xvrl** [comment]
:+1:

**Pull request merged**